src/Services/ConvertToQtiService.php

```diff
@@ -229,12 +210,14 @@ private function persistResultsFile(array $results, $outputFilePath)
         if ($this->dryRun) {
             return;
         }
-
         $this->output->writeln("\n<info>" . static::INFO_OUTPUT_PREFIX . "Writing conversion results: " . $outputFilePath . '.json' . "</info>\n");
 
         foreach ($results as $result) {
-            if (!empty($result['qti'])) {
-                file_put_contents($outputFilePath . '/' . $result['json']['reference'] . '.xml', $result['qti'][0]);
+
+            foreach ($result['json']['questions'] as $question) {
```

Comment:
Minor one - you could make the `if` check wrap the `foreach` block instead, since it will return the same result for every iteration and you could avoid entering the loop at all if it resolves to `false`.

Reply:
Done